Tools/TestWebKitAPI/Tests/IPC/IPCTestUtilities.h

```diff
@@ -32,6 +32,15 @@
 
 namespace TestWebKitAPI {
 
+template <typename T>
+std::optional<T> copyViaEncoder(const T& o)
```

Comment:
Could you just use this to create same-process copies of the Semaphore for the test code, and then not need to add the copy constructor?

It seems fine either way though, to support both in-process copies as well as IPC ones, just adds a bit more code that's only used for testing.

Reply:
> Could you just use this to create same-process copies of the Semaphore for the test code, and then not need to add the copy constructor?

While contrary to the goal of bringing order to (my) world, I think your suggestion is unfortunately the better approach.